docs/developer/releasing.md

```diff
@@ -94,17 +98,36 @@ For a new release branch, the release shepard will:
    fixes are available, cherry-picking the fixes into the release branch and
    starting a new release candidate.
 
-7. Create a PR to update code for the upcoming stable release. The changelog
-   sections for the release candidates should be replaced with a single section
-   for the stable release.
+   The `rc` version of the next release candidate would have to be incremented - 
+   e.g. from `v0.31.0-rc.0` to `v0.31.0-rc.1`.
+
+If no new issues were found during the environment testing, the shepherd can continue with 
+creating a **stable release**:
 
-8. Create the stable release tag.
+1. Create a new PR to [update code](#updating-code) using the _stable_
+   release version (i.e. without `-rc.N`).
 
-9. Force-push the `release` branch to point at the stable release tag. This
+2. Create the stable release tag.
+
+3. Force-push the `release` branch to point at the stable release tag. This
    branch is used to externally reference files in the repository for a stable
-   release.
+   release, e.g. the "[latest](https://grafana.com/docs/agent/latest/)" documentation.
+
+4. Make sure the documentation is updated:
+    1. A new version of the documentation should be visible on the 
+    [version switcher](https://grafana.com/docs/versions/?project=/docs/agent/).
+    2. The "[latest](https://grafana.com/docs/agent/latest/)" one should be identical to the latest version.
+    3. Check the "Upgrade guide" - the latest changes listed in it should be assigned to the latest version.
+
+5. [Publish the release on Github](#publishing-the-release)
+
+6. [Update Agent Operator Helm chart version](#update-agent-operator-helm-chart-version)
+
+7. Make sure [Homebrew is updated](#updating-homebrew).
 
-For patch releases, the release shepard will:
+8. Announce the release on community Slack channel.
+
+For **patch releases**, the release shepherd will:
```

Comment:
Hi @rfratto, do you think that the steps below are good enough to describe patch releases? If you agree, I could update the md file.

#### For **patch releases**, the release shepherd will:
1. Merge the fixes we need for the patch to `main`.
2. Raise a PR to `main` to do two things:
    1. Add a new section to the changelog for the patch release. **Move** the fixes listed in the changelog via step 1 from the "unreleased" section to the section for the new patch release.
    2. Update code so that `main` is referencing the new patch version in all files.
4. After the PR in step 2 is merged, cherry pick the commit from step 2 into the release branch.
5. Create a new tag for the patch release.

The above instructions have a few implications:
- When creating patch releases, there is no need for a release candidate.
- Changes made to patch releases are not listed in the changelog for the next stable version.
- The tag of the patch release does yet not exist at the time when we update code for the main branch to reference it.

NOTE: When patching an old release branch, we would do some things differently:
- `main` would not be updated to reference the patch version
- The change applied to the old release branch would have to be listed in two sections in the changelog:
        1. The patch release's section
        2. The "unreleased" section

Reply:
These look good 👍